### ISSUE TITLE
About Enter

### ISSUE THREAD

**Issue by Ab1eZz** [marktext/marktext#3482]
1. I press some Enters in a document. When I switch back to the document, I find that the Enters disappear.  Is it a bug or a feature?
2. Occasionally, I enter something in a document, then I click with the mouse or switch files. After that, what I just entered disappears. Only after I restart marktext will it be restored. But what i entered are still missing.

### Versions

- MarkText version: 0.17.1
- Operating system: Windows 10

**Comment by Jocs** [comment]
Thank you for reporting this. This issue was filed against MarkText 0.17.1 which used the legacy `muyajs` editor engine.

The root cause was in the old engine's `enterCtrl.js`: pressing Enter only called `partialRender()`, which did not dispatch the `stateChange`/`dispatchChange` event. As a result, new paragraphs created by Enter were rendered visually but not committed to the content-change pipeline — so switching tabs before any other action that did trigger `dispatchChange` (e.g. typing a character) could lose those changes.

MarkText has since migrated to a fully rewritten editor engine (`@muyajs/core`). In the new engine, every document mutation — including Enter key presses — goes through `JSONState.dispatch()`, which unconditionally emits a `json-change` event. The desktop renderer listens to this event and immediately updates the tab's saved markdown state, marks the tab dirty, and schedules auto-save if enabled. The scenario described in this issue is no longer possible in the current architecture.

Closing as resolved by the engine migration. If you are still seeing data loss on a recent build, please open a new issue with exact reproduction steps and the MarkText version.
